packages/blade/src/components/Input/PhoneNumberInput/__tests__/PhoneNumberInput.test.stories.tsx

```diff
@@ -118,20 +121,43 @@ ControlledState.play = async () => {
 
   // Type inside input
   await userEvent.clear(input);
+};
+
+export const ControlledStateTypeAndOnChange: StoryFn<
+  typeof PhoneNumberInput
+> = (): React.ReactElement => {
+  const [value, setValue] = React.useState('');
+  return (
+    <PhoneNumberInput
+      label={label}
+      value={value}
+      onChange={(e) => {
+        onChangeFn(e);
+        setValue(e.value);
+      }}
+    />
+  );
+};
+
+ControlledStateTypeAndOnChange.play = async () => {
+  const { getByLabelText } = within(document.body);
+
+  await sleep(100);
+  const input = getByLabelText(label);
+
   await userEvent.type(input, '1234567890');
 
-  // Ensure the value of the input updates
   await expect(input).toHaveValue('1234567890');
 
-  await expect(onChangeFn).toHaveBeenLastCalledWith(
```

Comment:
seems like toHaveBeenLastCalledWith is slow.. it was so slow that it was causing this test to timeout 💀

Reply:
interesting